fn/context_guard_test.go

```diff
@@ -439,3 +442,32 @@ func TestContextGuard(t *testing.T) {
 		}
 	})
 }
+
+// TestContextGuardCountGoroutines makes sure that ContextGuard doesn't create
+// any goroutines while waiting for contexts.
+func TestContextGuardCountGoroutines(t *testing.T) {
+	g := NewContextGuard()
```

Comment:
nit: add `t.Parallel()`

Reply:
t.Parallel() is not invoked in this test because it relies on an accurate count of active goroutines. Running other tests in parallel would introduce additional goroutines, leading to unreliable results.

I added a `NOTE` to the test.

Reply:
ah! makes sense 🙏